src/snapshot/download.js

```diff
@@ -43,19 +45,19 @@ function downloadImage(gd, opts) {
             reject(new Error('Sorry IE does not support downloading from canvas. Try {format:\'svg\'} instead.'));
         }
 
-        gd._snapshotInProgress = true;
+        _gd._snapshotInProgress = true;
```

Comment:
Won't this fail for the plain object case (`_gd === undefined`)?

Reply:
Good eye. This made me realize `Plotly.download` with figure objects as first argument was also broken. Fixed and :lock: ed in -> https://github.com/plotly/plotly.js/pull/2931/commits/a5af83c0275a6ae82c4b3cb9377982cc24cf268b